indico/cli/i18n.py

```diff
@@ -113,62 +244,246 @@ def _wrapper(**kwargs):
+    def decorator(fn):
+        fn = click.option('--python', is_flag=True, help='i18n used in python and Jinja code.')(fn)
+        fn = click.option('--react', is_flag=True, help='i18n used in react code.')(fn)
+        if require_js:
+            fn = click.option('--javascript', is_flag=True, help='i18n used in javascript code.')(fn)
+        else:  # the only command that does not require js is compile_catalog which checks the string format
+            fn = click.option('--no-check', is_flag=True, required=False,
+                              help='Skip running the string format validation.')(fn)
+        if require_locale:
+            fn = click.option('--locale', required=True)(fn)
+        if plugins:
+            fn = click.argument('plugin_dirs', required=True, type=click.Path(exists=True, file_okay=False,
+                                                                              resolve_path=True),
+                                nargs=-1, callback=_validate_plugin_dirs)(fn)
+        elif all_plugins:
+            fn = click.argument('plugin_dirs', type=click.Path(exists=True, file_okay=False, resolve_path=True),
+                                callback=_validate_all_plugins_dir)(fn)
+        return fn
+    return decorator
+
+
+def compile_catalog_react(directory=INDICO_DIR):
+    """Compile react catalogs."""
+    translations_dir = _get_translations_dir(directory)
+    try:
+        for locale in os.listdir(translations_dir):
+            po_file = os.path.join(translations_dir, locale, 'LC_MESSAGES', 'messages-react.po')
+            json_file = os.path.join(translations_dir, locale, 'LC_MESSAGES', 'messages-react.json')
+            if not os.path.exists(po_file):
+                continue
+            with _chdir(INDICO_DIR):
+                output = subprocess.check_output(['npx', 'react-jsx-i18n', 'compile', po_file], encoding='utf-8')
+            json.loads(output)  # just to be sure the JSON is valid
+            with open(json_file, 'w') as f:
+                f.write(output)
+    except subprocess.CalledProcessError as err:
+        click.secho('Error: ' + err, fg='red', bold=True, err=True)
+
+
+def extract_messages_react(directory=INDICO_DIR):
+    """Extract messagges for react."""
+    if directory is INDICO_DIR:
+        paths = [os.path.join('indico', 'web', 'client'), os.path.join('indico', 'modules')]
+    else:
+        packages = [x for x in find_packages(directory) if '.' not in x]
+        assert len(packages) == 1
+        client_path = os.path.join(directory, packages[0], 'client')
+        module_path = os.path.join(directory, packages[0], 'modules')
+        if not os.path.exists(client_path) and not os.path.exists(module_path):
+            return
+        paths = []
+        if os.path.exists(client_path):
+            paths.append(client_path)
+        if os.path.exists(module_path):
+            paths.append(module_path)
```

Comment:
Maybe a bit simpler?
```suggestion
        paths = []
        if os.path.exists(client_path):
            paths.append(client_path)
        if os.path.exists(module_path):
            paths.append(module_path)
        if not paths:
            return
```

Reply:
Okay